drivers/ieee802154/Kconfig.nrf5

```diff
@@ -31,4 +31,15 @@ config IEEE802154_NRF5_INIT_PRIO
 	  Set the initialization priority number. Do not mess with it unless
 	  you know what you are doing.
 
+config IEEE802154_NRF5_EXT_IRQ_MGMT
+	bool "Radio IRQ is managed by an external module"
+	help
+	  The driver may manage radio IRQs by itself, or use an external
+	  radio IRQ provider. When radio IRQs are managed by an external
+	  provider, the driver shall not interfere with it.
```

Comment:
The second part of the sentence is not clear; what does it mean "shall not interfere with it"? ('it', refers to the external provider). If this means, the driver shall not provide its own interrupt service routine, it could be written an such, IMHO

Reply:
How about `When radio IRQs are managed by an external provider, the driver shall not configure radio IRQs.` ?
The driver provides ISR, but such ISR is usually executed as a callback from a dispatcher in the external IRQ provider. I think it's not strictly ISR, but it is used like typical ISR.

Reply:
> The driver provides ISR, but such ISR is usually executed as a callback from a dispatcher in the external IRQ provider. I think it's not strictly ISR, but it is used like typical ISR.

I uderstand. It's more of a philosophical question, whether the callback can be called ISR (After all, it is executed in IRQ context..)

Reply:
> How about When radio IRQs are managed by an external provider, the driver shall not configure radio IRQs. ?

That's fine. I guess it is implicit that, in this case the ISR is not used (directly), but you could make it more clear by stating it:

Suggestion (optional)
`When radio IRQs are managed by an external provider, the driver shall not configure radio IRQs, and the driver's radio ISR is not to be used.
`

Reply:
Actually, there is another macro to handle direct ISRs: `NRF_802154_INTERNAL_RADIO_IRQ_HANDLING`. It is used only in the HAL (no need to expose it to shim, as in Zephyr the driver never defines ISR directly). I prefer to keep these macros separately and not suggest in this description that disabling this configuration option would driver's radio ISR would be used directly, as it's not true. ISR would be registered as callback in Zephyr kernel.